src/rotator_library/batched_persistence.py

```
@@ -0,0 +1,335 @@
+        lib_logger.debug(
+            f"BatchedPersistence started for {self._file_path.name} "
+            f"(interval={self._config.write_interval}s)"
+        )
+
+    async def _load_from_disk(self) -> None:
+        """Load state from disk file if it exists."""
+        if not self._file_path.exists():
+            return
+
+        try:
+            with open(self._file_path, "r", encoding="utf-8") as f:
```

Ruff (0.15.4)

[warning] Async functions should not open files with blocking methods like `open`

(ASYNC230)